instances/azp-build-asg/init.sh.tpl

```diff
@@ -15,9 +15,45 @@ hostnamectl set-hostname "$instance_id"
 
 azp_token=$(aws s3 cp s3://cncf-envoy-token/azp_token -)
 
-export AWS_DEFAULT_REGION=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | jq -r .region)
-aws ec2 replace-iam-instance-profile-association --iam-instance-profile Arn=${instance_profile_arn} \
-  --association-id $(aws ec2 describe-iam-instance-profile-associations --filter=Name=instance-id,Values=$instance_id | jq -r '.IamInstanceProfileAssociations[0].AssociationId')
+# Configured vars
+asg_name="${asg_name:-}"
+azp_pool_name="${azp_pool_name:-}"
+bazel_cache_bucket="${bazel_cache_bucket:-}"
+cache_prefix="${cache_prefix:-}"
+instance_profile_arn="${instance_profile_arn:-}"
+role_name="${role_name:-}"
```

Comment:
none of this is correct

these are template vars not env vars and this is a tpl not a sh

probs we do want to shellcheck it but either just manually or with some stubs to ignore stuff

Reply:
Ack, I am assuming we will revert a portion of this change before we go forward.